catalog/app/utils/Data.js

```diff
@@ -28,7 +28,13 @@ const reducer = Action.reducer({
 
 export function useData(request, params, { noAutoFetch = false } = {}) {
   // TODO: accept custom key extraction fn (params => key for comparison)
-  const [state, dispatch] = React.useReducer(reducer, initial)
+  const [state, setState] = React.useState(initial)
+  const stateRef = React.useRef()
+  stateRef.current = state
```

Comment:
I strongly recommend reduce or even remove state from `useData`, not add more

Reply:
well, i would be glad to do that, but how? "make things as simple as possible, but not simpler". seems like the problem requires some level of complexity and i dont see a way to simplify it further, however, i'm open to suggestions